installer.sh

```diff
@@ -0,0 +1,234 @@
+      TRAVIS_QUEUE_NAME="${1#*=}"
+      ;;
+    --skip_docker_populate=*)
+      SKIP_DOCKER_POPULATE="${1#*=}"
+      ;;
+    *)
+      printf "*************************************************************\n"
+      printf "* Error: Invalid argument.                                  *\n"
+      printf "* Valid Arguments are:                                      *\n"
+      printf "*  --travis_worker_version=x.x.x                            *\n"
+      printf "*  --docker_version=x.x.x                                   *\n"
+      printf "*  --travis_enterprise_host="demo.enterprise.travis-ci.com" *\n"
+      printf "*  --travis_enterprise_security_token="token123"            *\n"
+      printf "*  --travis_enterprise_build_endpoint="build-api"           *\n"
+      printf "*  --travis_queue_name="builds.linux"                       *\n"
+      printf "*  --skip_docker_populate=true                              *\n"
+      printf "*************************************************************\n"
+      exit 1
+  esac
+  shift
+done
+
+if [[ ! -n $DOCKER_VERSION ]]; then
+  export DOCKER_VERSION="17.12.0~ce-0~ubuntu"
+else
+  export DOCKER_VERSION
+fi
+
+if [[ ! -n $TRAVIS_WORKER_VERSION ]]; then
+  export TRAVIS_WORKER_VERSION="v3.5.0"
+else
+  export TRAVIS_WORKER_VERSION
+fi
+
+## We only want to run as root
+root_check() {
+  if [[ $(whoami) != "root" ]]; then
+    echo "This should only be run as root"
+    exit 1
+  fi
+}
+
+root_check
+##
+
+install_packages() {
+  apt-get install -y
+    apt-get install -y apt-transport-https \
+    ca-certificates \
+    curl \
+    software-properties-common \
+    jq
+}
+
+install_packages
+
+## Install and setup Docker
+docker_setup() {
+  : "${DOCKER_CONFIG_FILE:=/etc/default/docker}"
+
+  if [[ ! -f $DOCKER_APT_FILE ]]; then
+    curl -fsSL 'https://download.docker.com/linux/ubuntu/gpg' | apt-key add -
+    add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+  fi
+
+  apt-get update
+
+  if ! docker version &>/dev/null; then
+    apt-get install -y docker-ce=$DOCKER_VERSION
+  fi
+
+  jq -n '{"storage-driver": "overlay2", "icc": false, "log-driver": "journald"}' > /etc/docker/daemon.json
+  systemctl restart docker
+  sleep 2 # a short pause to ensure the docker daemon starts
+}
+
+docker_setup
+
+# Installs the travis-tfw-combined-env commandline tool
+install_travis_tfw_combined_env() {
+  curl -fsSL 'https://raw.githubusercontent.com/travis-ci/terraform-config/master/assets/tfw/usr/local/bin/travis-tfw-combined-env' > /usr/local/bin/travis-combined-env
+  chmod +x /usr/local/bin/travis-combined-env
+}
+
+install_travis_tfw_combined_env
+
+# Installs the wrapper script for running travis-worker
+install_travis_worker_wrapper() {
+  curl -fsSL 'https://raw.githubusercontent.com/travis-ci/terraform-config/master/assets/travis-worker/travis-worker-wrapper' > /usr/local/bin/travis-worker-wrapper
+  chmod +x /usr/local/bin/travis-worker-wrapper
+}
```

Comment:
I'm glad that these scripts are usable as-is!  What do you think about pinning to a tag or branch other than `master`?

Reply:
@meatballhat Definitely happy to pin something specific! Let's talk more about that & we'll pin versions to something specific in a follow-up PR.